docs/concepts.rst

```diff
@@ -956,6 +958,11 @@ procedure -- in complicated cases, it may not be able to figure out how to do
 this.  In these cases it will try to throw an exception rather than fail
 silently.) All future operations on the flow will reflect its updated status.
 
+To ensure that any functional changes are detected properly, you will need to
```

Comment:
wording nits:

- you have "To ensure that any functional changes are detected properly," twice; I'd drop the first one.
- "the versioning feature" can probably just be "versioning".
- :"functional changes": "code changes" might be clearer
- "may use stale cache": may want to be precise here and say something like "may keep using cached values from previous versions of the code"

Likewise for the docstring above.